src/mscorlib/src/System/Span.cs

```diff
@@ -152,5 +154,79 @@ public bool Equals(Span<T> other)
             return (_length == other._length) &&
                 (_length == 0 || JitHelpers.ByRefEquals(ref JitHelpers.GetByRef<T>(ref _rawPointer), ref JitHelpers.GetByRef<T>(ref other._rawPointer)));
         }
+
+        public bool TryCopyTo(Span<T> destination)
+        {
+            if (Length > destination.Length)
+                return false;
+
+            if (JitHelpers.ContainsReferences<T>())
+            {
+                var src = JitHelpers.GetByRef<T>(ref _rawPointer);
+                var dst = JitHelpers.GetByRef<T>(ref destination._rawPointer);
+
+                for (int i = 0; i < Length; i++)
+                    JitHelpers.AddByRef(ref dst, i) = JitHelpers.AddByRef(ref src, i);
+
+                return true;
+            }
+
+            unsafe
+            {
+#if WIN64
+                Buffer.Memmove((byte*)destination._rawPointer, (byte*)_rawPointer, (ulong)SizeInBytes);
+#else
+                Buffer.Memmove((byte*)destination._rawPointer, (byte*)_rawPointer, checked((uint)SizeInBytes));
+#endif
+                return true;
+            }
+        }
+
+        public bool TryCopyTo(T[] destination)
+        {
+            if (destination == null)
+                throw new ArgumentNullException("destination");
+
+            if (JitHelpers.ContainsReferences<T>())
+            {
+                var src = JitHelpers.GetByRef<T>(ref _rawPointer);
+                for (int i = 0; i < destination.Length; i++)
+                    destination[i] = JitHelpers.AddByRef(ref src, i);
```

Comment:
The copying for the `ContainsReferences` case can be also made faster by using memmoveGCRefs (it is what Array.Copy uses). But it can wait for later / separate PR.